src/gf.c

```diff
@@ -1220,7 +1220,17 @@ static void method_overwrite(jl_typemap_entry_t *newentry, jl_method_t *oldvalue
     jl_method_t *method = (jl_method_t*)newentry->func.method;
     jl_module_t *newmod = method->module;
     jl_module_t *oldmod = oldvalue->module;
-    if (newmod != jl_main_module || oldmod != jl_main_module) {
+
+    int isinteractive = 0;
+    if (!jl_generating_output()) {
+        jl_function_t *func = jl_get_function(jl_base_module, "isinteractive");
+        jl_value_t *ret = jl_call0(func);
+        JL_GC_PUSH1(&ret);
```

Comment:
Is a GC root needed in situations like this? I am not sure when the GC has the possibility of runnning.